terraform/main.tf

```diff
@@ -1,12 +1,12 @@
 terraform {
+  required_version = "~> 1.3"
```

Comment:
The implementation version restrictions further constrain the versions used when provisioning to avoid any surprises. This is different from the module level version constraint (see below)

The `~>` puts an upper bound restriction, where this upper bound restriction shown here is preventing from automatically adopting a major version change, but will readily accept a minor and/or patch version change

Reply:
@niallthomson no, the `~> 1.3` allows the rightmost digit to be anything larger than `3`.